python/ray/experimental/sgd/tf/tf_trainer.py

```diff
@@ -92,24 +95,65 @@ def __init__(self,
                 for i, worker in enumerate(self.workers)
             ])
 
-    def train(self):
+        import tensorflow as tf
+        self._Progbar = tf.keras.utils.Progbar
+
+    def _generic_model_driver(self, request_step, steps, metrics_prefix):
         """Runs a training epoch."""
+        progbar = self._Progbar(steps)
+
+        worker_stats = [None] * len(self.workers)
+        done = False
+        while not done:
+            reqs = [request_step(w) for w in self.workers]
+            for workerN, (type, logs) in enumerate(ray.get(reqs)):
```

Comment:
instead of `workerN`, can you do something like `worker_idx` or `worker_index`? Same goes for batch.